### PR TITLE
Add README.md for archived runtime packages

### PR DESCRIPTION
**Pull request by cston**
Fixes https://github.com/dotnet/corefx/issues/33170.

### CODE REVIEW

**Comment by cston** [comment]
cc @jaredpar, @stephentoub, @karelz, @danmosemsft

**Review by jaredpar — APPROVED** [review]

**Comment by ViktorHofer** [comment]
@ericstj should we consider harvesting (some of those) instead?

**Comment by jaredpar** [comment]
> should we consider harvesting (some of those) instead?

Harvesting?

File: src/libraries/Microsoft.CSharp/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Microsoft.CSharp package
+
+We are not accepting feature contributions to Microsoft.CSharp.
+The package is effectively archived.
```
Comment (ericstj) [inline src/libraries/Microsoft.CSharp/README.md]: Not sure if this is correct.  Archived implies no changes whatsoever.

**Comment by ericstj** [comment]
> Harvesting?

AKA redistributing old binaries in new packages.

> @ericstj should we consider harvesting (some of those) instead?

No, since the product still ships these and we support them we need to continue to build them and not extend the lifetime of past servicing branches past the product lifecycle.  We should only stop building things that are no longer part of the product.

File: src/libraries/Microsoft.CSharp/README.md (hunk 1 above)
Comment (jaredpar) [reply src/libraries/Microsoft.CSharp/README.md]: The phrase "effectively archived" is being applied to all of our code bases where we're only taking servicing level fixes. It's how we describe the old corefx, coreclr and core-setup repositories for example. These libraries fit into that theme.

File: src/libraries/System.Dynamic.Runtime/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# System.Dynamic.Runtime package
```
Comment (ericstj) [inline src/libraries/System.Dynamic.Runtime/README.md]: package here is a misnomer: we don't produce a package for this.  It's only in the shared framework.  Should this say "library" instead?  Same applies to System.Linq.Expressions.

File: src/libraries/Microsoft.CSharp/README.md (hunk 1 above)
Comment (ericstj) [reply src/libraries/Microsoft.CSharp/README.md]: I see.  It's up to you what language you use, but it came across as hyperbole to me when the following statement seemed to contradict it.  To me, "effectively archived" makes sense for repositories where the code has moved somewhere else for the latest release, it makes less sense when the library is still building out of the codebase and you've just racheted up the bar: nothing has been archived it just has a higher bar.

**Review by stephentoub — APPROVED** [review]

**Comment by ltrzesniewski** [comment]
I'd still like https://github.com/dotnet/corefx/issues/16287 to be fixed in System.Linq.Expressions since it's a regression from the .NET Framework. Would you be willing to consider this?

**Comment by jaredpar** [comment]
@ltrzesniewski 

Agree with @danmosemsft comment there:

> However bringing back a method that was previously present in desktop may fall under the "regressions" category described in that issue. 

If there was a solid plan for how to bring this back on top of the feature set of .NET Core (given the old emit technology wasn't ported to desktop) then we'd be willing to consider a pull request for that.

**Pull request merged**